lib/logger/renovate-logger.ts

```diff
@@ -0,0 +1,128 @@
+import is from '@sindresorhus/is';
+import type * as bunyan from 'bunyan';
+import { once, reset as onceReset } from './once';
+import { getRemappedLevel } from './remap';
+import type { Logger } from './types';
+import { getMessage, toMeta, withSanitizer } from './utils';
+
+const loggerLevels: bunyan.LogLevelString[] = [
+  'trace',
+  'debug',
+  'info',
+  'warn',
+  'error',
+  'fatal',
+];
+
+type loggerFunction = (p1: string | Record<string, any>, p2?: string) => void;
+
+export class RenovateLogger implements Logger {
+  logger: Logger = { once: { reset: onceReset } } as any;
+
+  constructor(
+    private readonly bunyanLogger: bunyan,
+    private context: string,
+    private meta: Record<string, unknown>,
+  ) {
+    for (const level of loggerLevels) {
+      this.logger[level] = this.logFactory(level) as never;
+      this.logger.once[level] = this.logOnceFn(level);
+    }
+  }
+
+  addStream(stream: bunyan.Stream): void {
+    this.bunyanLogger.addStream(withSanitizer(stream));
+  }
+
+  childLogger(): RenovateLogger {
+    return new RenovateLogger(
+      this.bunyanLogger.child({}),
+      this.context,
+      this.meta,
+    );
+  }
+
+  trace = this.log.bind(this, 'trace');
+  debug = this.log.bind(this, 'debug');
+  info = this.log.bind(this, 'info');
+  warn = this.log.bind(this, 'warn');
+  error = this.log.bind(this, 'error');
+  fatal = this.log.bind(this, 'fatal');
+
+  once = this.logger.once;
```

Comment:
```suggestion
  readonly trace = this.log.bind(this, 'trace');
  readonly debug = this.log.bind(this, 'debug');
  readonly info = this.log.bind(this, 'info');
  readonly warn = this.log.bind(this, 'warn');
  readonly error = this.log.bind(this, 'error');
  readonly fatal = this.log.bind(this, 'fatal');

  readonly once = this.logger.once;
```

I don't like those, i would prefer to use wrapper functions.